src/lib/core/CHIPTLV.h

```diff
@@ -849,7 +849,7 @@ class DLL_EXPORT TLVWriter
      * @param[in]   maxLen  The maximum number of bytes that should be written to the output buffer.
      *
      */
-    void Init(uint8_t * buf, uint32_t maxLen);
+    void Init(uint8_t * buf, size_t maxLen);
```

Comment:
GetLengthWritten() also needs to be changed if possible.

Reply:
That entrains a bunch of API changes various places, sadly.  Filed https://github.com/project-chip/connectedhomeip/issues/8953